docs/architecture/adr-069-unordered-account.md

```diff
@@ -0,0 +1,177 @@
+# ADR 070: Un-Ordered Transaction Inclusion
+
+## Changelog
+
+* Dec 4, 2023: Initial Draft
+
+## Status
+
+Proposed
+
+## Abstract
+
+We propose a way to do replay-attack protection without enforcing the order of transactions, it don't use nonce at all. In this way we can support un-ordered transaction inclusion.
+
+## Context
+
+As of today, the nonce value (account sequence number) prevents replay-attack and ensures the transactions from the same sender are included into blocks and executed in sequential order. However it makes it tricky to send many transactions from the same sender concurrently in a reliable way. IBC relayer and crypto exchanges are typical examples of such use cases.
+
+## Decision
+
+We propose to add a boolean field `unordered` to transaction body to mark "un-ordered" transactions.
+
+Un-ordered transactions will bypass the nonce rules and follow the rules described below instead, in contrary, the default ordered transactions are not impacted by this proposal, they'll follow the nonce rules the same as before.
+
+When an un-ordered transaction are included into block, the transaction hash is recorded in a dictionary, new transactions are checked against this dictionary for duplicates, and to prevent the dictionary grow indefinitly, the transaction must specify `timeout_height` for expiration, so it's safe to removed it from the dictionary after it's expired.
```

Comment:
```suggestion
When an un-ordered transaction is included into a block, the transaction hash is recorded in a dictionary. New transactions are checked against this dictionary for duplicates, and to prevent the dictionary grow indefinitely, the transaction must specify `timeout_height` for expiration, so it's safe to removed it from the dictionary after it's expired.
```